src/Compilers/Core/CodeAnalysisTest/ShadowCopyAnalyzerAssemblyLoaderTests.cs

```diff
@@ -0,0 +1,105 @@
+
+            Assert.NotNull(analyzerDependencyReference.GetAssembly());
+
+            TempFile CreateAnalyzerDependency()
+            {
+                var analyzerDependencySource = @"
+using System;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+public abstract class AbstractTestAnalyzer : DiagnosticAnalyzer
+{
+    protected static string SomeString = nameof(SomeString);
+    public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get { throw new NotImplementedException(); } }
+    public override void Initialize(AnalysisContext context) { throw new NotImplementedException(); }
+}";
+
+                var analyzerDependencyCompilation = CSharp.CSharpCompilation.Create(
+                   "AnalyzerDependency",
+                   new SyntaxTree[] { CSharp.SyntaxFactory.ParseSyntaxTree(analyzerDependencySource) },
+                   new MetadataReference[]
+                   {
+                    TestReferences.NetStandard20.NetStandard,
+                    TestReferences.NetStandard20.SystemRuntimeRef,
+                    MetadataReference.CreateFromFile(immutable.Path),
+                    MetadataReference.CreateFromFile(microsoftCodeAnalysis.Path)
+                   },
+                   new CSharp.CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary));
+
+                return directory.CreateDirectory("AnalyzerDependency").CreateFile("AnalyzerDependency.dll").WriteAllBytes(analyzerDependencyCompilation.EmitToArray());
+            }
+
+            TempFile CreateMainAnalyzerWithDependency(TempFile analyzerDependency)
+            {
+                var analyzerMainSource = @"
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public sealed class TestAnalyzer : AbstractTestAnalyzer
+{
+    private static string SomeString2 = AbstractTestAnalyzer.SomeString;
+}";
+                var analyzerMainCompilation = CSharp.CSharpCompilation.Create(
+                   "AnalyzerMain",
+                   new SyntaxTree[] { CSharp.SyntaxFactory.ParseSyntaxTree(analyzerMainSource) },
+                   new MetadataReference[]
+                   {
+                    TestReferences.NetStandard20.NetStandard,
+                    TestReferences.NetStandard20.SystemRuntimeRef,
+                    MetadataReference.CreateFromFile(immutable.Path),
+                    MetadataReference.CreateFromFile(microsoftCodeAnalysis.Path),
+                    MetadataReference.CreateFromFile(analyzerDependency.Path)
+                   },
+                   new CSharp.CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary));
+
+                return directory.CreateDirectory("AnalyzerMain").CreateFile("AnalyzerMain.dll").WriteAllBytes(analyzerMainCompilation.EmitToArray());
+            }
+        }
+    }
+}
+
+#endif
```

Comment:
Consider adding the following:

```
#else
#error unsupported configuration
#endif
```

This helps guard against future changes to the target frameworks that roslyn uses. Consider that in the near future netcoreapp3.0 will be a valid target framework. With this proposed change we would get a compiler error if we forgot to update this file. Without the test would just silently stop running. 

I've been trying to take this approach in all places where we have `#if` as I've gotten this wrong several times.

Reply:
Thanks, done.